pkg/controllers/clusterresourceplacement/placement_status_test.go

```diff
@@ -954,16 +962,219 @@ func TestSetPlacementStatus(t *testing.T) {
 			if err := r.setPlacementStatus(context.Background(), crp, selectedResources, tc.latestPolicySnapshot, tc.latestResourceSnapshot); err != nil {
 				t.Fatalf("setPlacementStatus() failed: %v", err)
 			}
-			statusCmpOptions := []cmp.Option{
-				// ignore the message as we may change the message in the future
-				cmpopts.IgnoreFields(metav1.Condition{}, "Message", "LastTransitionTime"),
-				cmpopts.SortSlices(func(c1, c2 metav1.Condition) bool {
-					return c1.Type < c2.Type
-				}),
-			}
+
 			if diff := cmp.Diff(tc.wantStatus, &crp.Status, statusCmpOptions...); diff != "" {
 				t.Errorf("buildPlacementStatus() status mismatch (-want, +got):\n%s", diff)
 			}
 		})
 	}
 }
+
+func TestBuildFailedResourcePlacements(t *testing.T) {
+	tests := map[string]struct {
+		work          *fleetv1beta1.Work
+		wantIsPending bool
+		wantRes       []fleetv1beta1.FailedResourcePlacement
+	}{
+		"pending if not applied": {
+			work:          &fleetv1beta1.Work{},
+			wantIsPending: true,
+			wantRes:       nil,
+		},
+		"No resource if applied successfully": {
+			work: &fleetv1beta1.Work{
+				ObjectMeta: metav1.ObjectMeta{
+					Generation: 1,
+				},
+				Status: fleetv1beta1.WorkStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:               fleetv1beta1.WorkConditionTypeApplied,
+							ObservedGeneration: 1,
+							Status:             metav1.ConditionTrue,
+						},
+					},
+				},
+			},
+			wantIsPending: false,
+			wantRes:       nil,
+		},
+		"pending if applied not on the latest generation": {
+			work: &fleetv1beta1.Work{
+				ObjectMeta: metav1.ObjectMeta{
+					Generation: 2,
+				},
+				Status: fleetv1beta1.WorkStatus{
+					Conditions: []metav1.Condition{
+						{
+							Type:               string(fleetv1beta1.WorkConditionTypeApplied),
+							ObservedGeneration: 1,
+							Status:             metav1.ConditionTrue,
+						},
+					},
+				},
+			},
+			wantIsPending: true,
+			wantRes:       nil,
+		},
+		"pending if applied failed with multiple object": {
```

Comment:
Hi Ryan! Just a nit, but this should be something like "report failure if applied failed with multiple" object? The case below also specifies "pending" but is actually of a failure state, if I understand it correctly.